base/dates/io.jl

```diff
@@ -159,14 +159,19 @@ function parse(x::AbstractString,df::DateFormat)
         throw(ArgumentError("Delimiter mismatch. Couldn't find first delimiter, \"$(df.slots[1].transition)\", in date string"))
     end
     periods = Period[]
-    extra = []
+    extra = Any[]  # Supports custom slot types such as TimeZone
```

Comment:
Doesn't actually change anything. I just wanted to be explicit that we can't be sure as to what types will be added into `extra`.